### DISCUSSION TITLE
tutorials/scripting/gdextension/gdextension_file

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#266] [User-contributed notes]
# tutorials/scripting/gdextension/gdextension_file

Introduction: The.gdextension file in your project contains the instructions for how to load the GDExtension. The instructions are separated into specific sections. This page should give you a quic...

https://docs.godotengine.org/en/stable/tutorials/scripting/gdextension/gdextension_file.html

**Comment by markeel** [comment]
Note that if the extension only has entries for the *build* as "editor", it will not be exported because that isn't one of the feature tags normally used for exporting.  In 4.3 this is silent but in 4.4 a warning will appear.  If it is really an editor only extension, then simply exclude the extension in the Resources tab of the export dialog during export.

For example the Git plugin is editor-only and has something like:
```
macos.editor = "...dylib"
windows.editor.x86_64 = "...dll"
linux.editor.x86_64 = "...so"
linux.editor.arm64 = "...so"
linux.editor.rv64 = ""
```
and so cannot be exported to a linux platform for a non-editor build (and really makes no sense to), so it requires the project owner to update Resources to identify that plugin to be excluded.

**Comment by TheOathMan** [comment]
Can i comment off something in the .gdextension file ? Something equivalent to ## in gdscript.

**Reply by Ivorforce** [reply]
Yes, I believe comments can be added with a semicolon:

```ini
; Test test
macos.x86_64.single.debug = "./macos/numdot.macos.template_debug.x86_64.framework/numdot.macos.template_debug.x86_64"
; [...]
```

**Reply by Calinou** [reply]
This should work as the TSCN/TRES/ConfigFile formats allow comments starting with `;`, similar to INI.

**Reply by Calinou** [reply]
Pull request opened:

- https://github.com/godotengine/godot-docs/pull/11381

**Comment by AndreasOM** [comment]
The .gdextension file supports relative paths for libraries and DLLs, which makes installation much more flexible for users.
However, most official examples only show res:// paths.
It should be clearly mentioned that relative paths are fully supported.
For reference, even the official C++ extension template uses relative paths:
https://github.com/godotengine/godot-cpp-template/blob/main/demo/bin/example.gdextension

And yes, I just did spent two hours banging my head on the keyboard finding any official confirmation that relative paths are working.

**Reply by Calinou** [reply]
Pull request opened:

- https://github.com/godotengine/godot-docs/pull/11381

**Comment by Buckram123** [comment]
Note that depending on another GDExtension is currently mostly unimplemented. While you may successfully call methods belonging to another extension, extending classes defined in other GDExtensions is not supported. There is currently no deterministic way to define the extension load order, which is handled internally by `.godot/extension_list.cfg` and if the dependent extension is loaded before its dependency, it will lead to initialization errors. On top of that the `dependencies` section defined here is strictly for external dynamic libraries and managed by your GDExtension and should not be used to declare a dependency on another GDExtension.

It's in the priorities list https://godotengine.org/priorities/#allow-gdextensions-to-communicate-with-one-another but I couldn't find an actual issue or PR
